### PR TITLE
Implement support for Google Cloud Storage in CP Java API

### PR DESCRIPTION
**Pull request by ekazachkova**
Implements issue #10 . Provides apply storage policy method.

### CODE REVIEW

File: api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java [hunk 1]
```diff
@@ -69,20 +79,26 @@
...
+    private static final String LIFECYCLE_FIELD = "lifecycle";
+    private static final String GOOGLE_STORAGE_API_URL = "https://www.googleapis.com/storage/";
 
     private static final byte[] EMPTY_FILE_CONTENT = new byte[0];
     private static final Long URL_EXPIRATION = 24 * 60 * 60 * 1000L;
 
     private final MessageHelper messageHelper;
     private final GCPRegion region;
     private final GCPClient gcpClient;
+    private GoogleStorageRestApiClient storageRestApiClient;
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java]: Why do we need as field and not local variable?

File: api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java [hunk 2]
```diff
@@ -473,4 +544,53 @@ private void deleteBucket(final String bucketName, final Storage client) {
...
+            executeRequest(() ->
+                    storageRestApiClient.disableLifecycleRules(buildToken(token), LIFECYCLE_CONTENT_TYPE,
+                            bucketName, LIFECYCLE_FIELD, EMPTY_LIFECYCLE_CONTENT));
+        } catch (IOException e) {
+            throw new DataStorageException(String
+                    .format("Failed to disable lifecycle rules for bucket %s", bucketName), e);
+        }
+    }
+
+    private <T> T executeRequest(Supplier<Call<T>> request) {
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java]: I'd suggest to move all logic for creating client and executing request to `GoogleStorageRestApiClient` to a separate class

File: api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java (hunk 1 above)
Comment (ekazachkova) [reply api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java]: done

File: api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java (hunk 2 above)
Comment (ekazachkova) [reply api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java]: done

**Pull request merged**